### PR TITLE
Use macro to generate SQL parsers

### LINKED ISSUE #909: [Macro] Implement the `parse_*` fn generation macro
### Background 
Currently, we defined `parse_*` fn separately, which shows similar pattern; 1) tokenize 2) parse tokens with specific sql-parser fn.
By introducing new decl macro which takes `fn_name`, `parse_fn_name`, `output type` as args, `parse_*` functions can be easily generated.  
https://github.com/gluesql/gluesql/blob/10824e0c567cbd2689079ae771f59339fee0e581/core/src/parse_sql.rs#L41-L59

### TODO
1. impl parse_* macro
2. replace code blocks with macro

### PR DESCRIPTION
**Pull request by sa1**
Fixes #909

### CODE REVIEW

File: core/src/parse_sql.rs [hunk 1]
```diff
@@ -18,124 +18,55 @@ pub fn parse<Sql: AsRef<str>>(sql: Sql) -> Result<Vec<SqlStatement>> {
...
+parse_!(parse_order_by_expr, parse_order_by_expr, SqlOrderByExpr);
+parse_!(
+    parse_order_by_exprs,
+    parse_comma_separated,
+    parse_order_by_expr,
+    Vec<SqlOrderByExpr>
+);
+parse_!(parse_column_def, parse_column_def, SqlColumnDef);
+parse_!(parse_data_type, parse_data_type, SqlDataType);
+parse_!(parse_sql_assignment, parse_assignment, SqlAssignment);
```
Comment (ever0de) [inline core/src/parse_sql.rs:71]: In this case, it seems that there is no need to enter sql in the middle, so I think you can delete it.

**Review by ever0de — COMMENTED** [review]
Welcome

I left my little comments. This is a formatting level opinion, so please leave any additional comments.

**Comment by sa1** [comment]
Thanks, I implemented those changes.

**Review by panarch — APPROVED** [review]
Looks quite much better! Thanks a lot 👍

**Pull request merged**